pkg/backend/openshiftcluster.go

```diff
@@ -185,6 +186,11 @@ func (ocb *openShiftClusterBackend) handle(ctx context.Context, log *logrus.Entr
 
 		stop()
 
+		// This Sleep ensures that the monitor has enough time
+		// to capture the deletion (by reading from the changefeed)
+		// and stop monitoring the cluster.
+		// TODO: Provide better communication between RP and Monitor
+		time.Sleep(time.Until(t.Add(time.Second * 20)))
```

Comment:
Adding 20 seconds to _every_ cluster delete is not a great customer experience to introduce for this edge case. Ideally we would follow the [recommended delete approach of using a TTLs](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/change-feed-design-patterns#deletes), 10 seconds in our case, but due to use-cases where users are creating and tearing down clusters using the same name/RG/VNET, etc, we may run into issues when the TTL has not expired yet.

To limit the effect of this change, wouldn't it make sense to add a `deleteStartedAt := time.Now()` variable before `err = m.Delete(ctx)`, and then only sleep here if the delete has taken less than 10 seconds? Or 15 to be absolutely certain.

Reply:
Ignore my first comment. I see that you have added a conditional sleep on the deletion start time.

Reply:
Ignore my previous comment, I see you have added a conditional sleep using the deletion start time.